common/forks/forks_test.go

```diff
@@ -64,3 +69,64 @@ func TestBlocksUntilFork(t *testing.T) {
 		})
 	}
 }
+
+func TestBlockRange(t *testing.T) {
+	tests := []struct {
+		name         string
+		forkHeight   uint64
+		forkHeights  []uint64
+		expectedFrom uint64
+		expectedTo   uint64
+	}{
+		{
+			name:         "ToInfinite",
+			forkHeight:   300,
+			forkHeights:  []uint64{100, 200, 300},
+			expectedFrom: 300,
+			expectedTo:   math.MaxUint64,
+		},
+		{
+			name:         "To300",
+			forkHeight:   200,
+			forkHeights:  []uint64{100, 200, 300},
+			expectedFrom: 200,
+			expectedTo:   300,
+		},
+		{
+			name:         "To200",
+			forkHeight:   100,
+			forkHeights:  []uint64{100, 200, 300},
+			expectedFrom: 100,
+			expectedTo:   200,
+		},
+		{
+			name:         "To100",
+			forkHeight:   0,
+			forkHeights:  []uint64{100, 200, 300},
+			expectedFrom: 0,
+			expectedTo:   100,
+		},
+		{
+			name:         "To200-1",
+			forkHeight:   100,
+			forkHeights:  []uint64{100, 200},
+			expectedFrom: 100,
+			expectedTo:   200,
+		},
+		{
+			name:         "to2",
+			forkHeight:   1,
+			forkHeights:  []uint64{1, 2},
```

Comment:
Assuming `forkHeights` is the result of `CollectSortedForkHeights`, `forkHeights` won't have duplicates.

Reply:
Makes sense. But then this won't work in the general case, e.g. if the prover has version `london`, the correct range should be `[0, 0)` but it will return `[0, 100)`.

But that's not an issue in practice. In fact, I think we should remove all irrelevant forks from `CollectSortedForkHeights`, so for now we should only keep `shanghai` and `banach`.

Reply:
yes, I assume this won't happen. Because the sql will like this `select .... where start_block_number >= 0 and end_blcok_number <0`.

Reply:
No, what I mean, is this edge case:

- Example: `londonBlock = 0, shanghaiBlock = 0`.
- Prover provides `forkName = "london"`.
- `BlockRange` should return `0, 0` but it will incorrectly return `0, maxUint`.

Reply:
Yeah, currently, it return `0, maxUnit`. Let me check whether need update it

Reply:
```
func BlockRange(forkHeight uint64, forkHeights []uint64) (from, to uint64) {
	length := len(forkHeights)
	if length == 0 {
		return 0, 0 // 处理空切片的情况
	}

	to = math.MaxUint64

	for i, height := range forkHeights {
		if forkHeight == height {
			from = height
			// 当 forkHeight 等于 height 且不是数组最后一个元素时
			if i < length-1 {
				to = forkHeights[i+1]
			} else {
				// 当 forkHeight 等于数组最后一个元素时
				to = math.MaxUint64
			}
			return
		} else if forkHeight < height {
			to = height
			if i != 0 {
				from = forkHeights[i-1]
			} else {
				from = 0
			}
			return
		}
		from = height
	}
	// 当 forkHeight 大于数组中所有元素时
	if forkHeight > forkHeights[length-1] {
		from = forkHeights[length-1]
	}
	return
}
```

Can run success all the case, but it's too complicated. And forkHeights don't have duplicated number. So How about  check forkHeights can't have duplicated number?

Reply:
We currently do have duplicates, so this check will always break.

Reply:
ok, rollback it